### DISCUSSION TITLE
absolute beginners

### DISCUSSION THREAD

**Discussion by ghost** [helge17/tuxguitar#593] [General]
first things first: it is GREAT that you are working on tuxguitar again! I am so glad and graceful -although I only use it to convert midi to TABS - just re-learning guitar. My issues: I am rather blind and need really BIG fonts and so on; did not manage to find the best format. and a dummy question; would it be possible to generate fingering while converting a midi file to TAB?  you see: At the moment I use a very low level of tux guitar --but for sure it motivates and helps me!

**Comment by guiv42** [comment]
> Im rather blind and need really BIG fonts and so on; did not manage to find the best format.

Could you please give a list of all elements  you would like to be able to magnify? Or more generally, any suggestion that could make it more comfortable for you? I won't promise anything, but it could be worth a try.

> would it be possible to generate fingering while converting a midi file to TAB? 

I'm assuming you're talking about *left hand* fingering (for a right-handed guitar), as there is currently  no concept of right hand fingering in TuxGuitar.
This has been [discussed already](https://github.com/helge17/tuxguitar/issues/409#issuecomment-2156388917), it's a complex topic. I have already improved this midi import feature last year, at least now there are no more negative fret numbers ;)
But no, I have no solution to propose (yet). My only recommendation is to look for *tablature* files on the Internet when you want to practice a song, there are many resources available out there including free ones. Midi is not done for this use case.

**Reply by ghost** [reply]
Hi- you're right; at the moment I do not know tuxguitar nor guitarplaying good enough t ounderstand deeper the possibilities and limits. Scaling: if I maximize the fonts, the numbers collide, the position on lines is difficult to identify; so I wonder if it would be possible to scale the complete TAB (as ZOOM can do it on screen) for printing.  Do you know what I mean? Actually I go to a copyshop and print A3 from A4- it is okay to do so; if it would be possible to insert line breaks and scaled TAB- I could stay at home :-)

**Reply by guiv42** [reply]
OK, so you're printing the tabs.
> if it would be possible to insert line breaks and scaled TAB- I could stay at home :-)

Line breaks are already implemented in tuxguitar-next branch, and will be available at next major release.
I have no idea of the complexity to enable scaling when printing / exporting to pdf. I'll have a look.

Edit: it looks pretty easy :)

**Comment by guiv42** [comment]
zoom control for [print / print preview / export to pdf]: implemented, will be available in next *minor* release
line break: implemented, will be available in next *major* release (requires file format update)

**Reply by ghost** [reply]
Hi! Wow, you ŕe really busy - looks good. ----But: either I am still VERY poor in playing or there are wired outputs in TABs. Please take a look on the attached file; a simple piece....Please don`t mind the red colour...... Greetings!
[Krieger_Menuett_in_A-Moll.pdf](https://github.com/user-attachments/files/18141752/Krieger_Menuett_in_A-Moll.pdf)

**Reply by guiv42** [reply]
Hi. I don't really understand your last question.
However, I had a quick look at the attached pdf file, and it shows quite a few problems. Where did you get this tab from?

If I should bet... I would say you imported a MIDI file with TuxGuitar. Is that correct?

There are many notes tied to nothing at the beginning of measures. This is a typical symptom when importing a file where timestamps of MIDI events are not perfectly aligned on the defined tempo.
I don't recommend working with MIDI files for TuxGuitar. As it is described in user doc, do not expect magical results when importing MIDI, this file format aims at being played by a synthesizer, by no means it is precise enough to write a tablature.

**Reply by ghost** [reply]
Hi -ohh THAT IS sad (for me), because TABS are so far the only possibility for me to practice, and I was happy about the ability to import MIDI and get TABS. Well, classical music seems to keep a matter for elites ---or top-notchers.  I*ll gonna become a rockstar playing 3 accords---- urrgh. Nevermind . thank you and good luck!

**Reply by guiv42** [reply]
I don't know what kind of tabs you're looking for, but many resources are available on the Internet under Guitar Pro format, even free ones.

**Reply by ghost** [reply]
Hi! Yes thanks, I was looking for classical TABS and there are some rather oldfashioned sites that provides those TABS.
 Not sure if it is interesting for you: the old tuxguitar (in my case it is listed as version 1.5.6+dfsg1-6) transferred MIDI to TABS without those errors  that occured in converting by the new version. 
It is great that you keep tuxguitar alive - furthermore good luck and success! All the best wishes, Leo

**Reply by guiv42** [reply]
To be honest it's a bit disappointing: I already spent some time fixing old bugs in MIDI import function (e.g. negative fret numbers, or lost notes). If the same midi file gives better results with 1.5.6 don't hesitate to create an issue. Just take care to attach your midi file so I can analyze (or else I won't be able to do anything).

**Reply by belfie13** [reply]
oh no, all that hard work, i'll look into it and try make sense of whats happening here for you

**Reply by belfie13** [reply]
@leo-22-afk can you list some sites where you get your midi from or specific files i can look at thanks

**Reply by ghost** [reply]
OH- I did not intend to discourage you!!! Please remember: I am a newbie in guitar playing and a NOOB considering development - so my needs are very special and not soooo relevant for an app like tuxguitar. Your work is fine, and I can help myself by using the old tuxguitar - fits to me ....
A midi- and TAB source is https://braumeister.tripod.com/. That Menuett-TAB I sent to you was converted by new tuxguitar. 
To be sure that my mournings won`t hurt you anymore I`ll stay calm and practice playing guitar - have a peaceful time and all the best! Leo

**Reply by belfie13** [reply]
no no problem at all, i use tuxguitar for midi import/export mainly as well and we had a few issues that were worked through and the string/fret selection was one that should have been fixed

**Comment by guiv42** [comment]
Hi @leo-22-afk 
First, I can confirm @belfie13 's point of view: there is no problem at all, and no one got discouraged.

> so my needs are very special and not soooo relevant for an app like tuxguitar

I would not say so. TuxGuitar aims at helping guitarists to learn guitar. And for sure this includes beginners! (who needs to learn?).
Whenever you encounter problems with TuxGuitar, do not hesitate to create a discussion or an issue. You may not be alone with your problem, and if we can solve it this would benefit the whole community.

Coming back to your specific tab, I tried to import it with both TuxGuitar 1.5.6 and 1.6.5, and I would not say one is better than the other.
I confirm you cannot see those weird notes, tied to nothing with 1.5.6. But I do believe it's just because of a bug in 1.5.6, which just ignores some notes even if they are present in the original midi file.

**Reply by belfie13** [reply]
I might add that the `2024-12-17 next macosx jfx` version has different outcomes when comparing `shift up/down string` with the _keyboard shortcut_ and the _right click menu_. the _shortcut_ has trouble with some notes, shifting them up or down multiple strings. `swt` doesn't have that problem.

easy to replicate I hope, create a new project and add a chord say:
```
---
-5-
---
-9-
-7-
---
```
and try to `shift up/down strings` with the **shortcut keys** then with the **right click menu** or the **application menu** and you will see

might be linked to our issue?

@leo-22-afk what's the exact version you're having trouble with?
Edit: can just copy the file name for example `tuxguitar-2024-12-17-tuxguitar-next-macosx-jfx-cocoa-x86_64.app` thanks

**Reply by guiv42** [reply]
@belfie13 I think the problem you mention is completely unrelated. Where exactly is the inconsistency between menu and shortcut? (A small video would help). If you confirm there is a problem please create an independent issue.

**Comment by guiv42** [comment]
I had a quick look at the midi file, and I can explain why it is incorrectly imported by TuxGuitar.
Two reasons for that:
- timing defined by the midi events is **not** precise
  Example: first E (a quarter according to pdf version of the file) has a duration of 231 ticks, but second quarter starts at tick 256. This is clearly inconsistent, and TuxGuitar cannot do anything to fix this. To be more clear: the midi file does **not** define this note as a quarter...
- there are several voices in the track: 2 first notes start at the same tick 0, but do not end at the same tick. TuxGuitar never implemented multi-voice import

See attachment: midi file, TuxGuitar files after importing with 1.5.6 and 1.6.5, and pdf
[Krieger_Menuett_in_A-Moll.tar.gz](https://github.com/user-attachments/files/18201598/Krieger_Menuett_in_A-Moll.tar.gz)

In first beat, the A is supposed to be a dotted half. It's interpreted erroneously as a quarter by both versions of TuxGuitar.
Then, the missing part of this note is (erroneously) forgotten by TG1.5.6, but it reappears (erroneously again) in measure 2 with TG1.6.5.
So: both versions are clearly wrong. But 1.6.5 is less readable, see issue #612

**Comment by guiv42** [comment]
@leo-22-afk , I have good news: I could fix the tied notes issue when importing multi-voices midi file (#613)
It will take some time before it is integrated into a new release, so please be patient.
Until then, here is the result of midi file import after the fix is implemented, for the song you are currently working.
[Krieger_Menuett_in_A-Moll_fix.zip](https://github.com/user-attachments/files/18203617/Krieger_Menuett_in_A-Moll_fix.zip)

Timing is not fully precise, but as I mentioned above the midi file itself *is imprecise*.
Have fun ;)

**Reply by ghost** [reply]
Thank you so much - really. For now I'd like to share an image instead of mourning.... ALL THE BEST! Leo
(I thought we were going out)
![no longer ADVENT time](https://github.com/user-attachments/assets/20569e0a-8c04-45e7-bd98-cdd585202ef8)
